invenio/modules/workflows/models.py

```diff
@@ -682,16 +727,15 @@ def change_status(self, message):
 
     def get_current_task(self):
         """Return the current task from the workflow engine for this object."""
-        extra_data = self.get_extra_data()
         try:
-            return extra_data["_task_counter"]
+            return self.extra_data["_task_counter"]
         except KeyError:
             # Assume old version "task_counter"
-            return extra_data["task_counter"]
+            return self.extra_data["task_counter"]
```

Comment:
>`task_counter`

Yes, it would be a good idea for the model to natively understand the `MachineState` so that we can load and dump it easily.

>`action`

If this is only used by the HoldingPen, I think it should either be in a dedicated model or we should structure `extra_data` better, eg:

```python
extra_data['holdingpen']['action']
```